test/should_pass/return_fun.erl

```diff
@@ -14,17 +17,5 @@ return_fun_union() -> fun nil/0.
 -spec return_fun_any_arity() -> fun((...) -> list()).
 return_fun_any_arity() -> fun nil/0.
 
--spec return_fun_intersection() -> fun((any()) -> integer()).
-return_fun_intersection() -> fun number/1.
```

Comment:
I know you just moved this function around, but why should this pass? The spec says that `return_fun_intersection/0` returns a function that returns an integer, but it actually returns a function that can also return a float (if I pass it a float). A function that might return a float should not be a subtype of a function that returns an integer.